docroot/profiles/dv/modules/custom/activity/activity_logger/activity_logger.tokens.inc

```diff
@@ -148,16 +158,54 @@ function activity_logger_tokens($type, $tokens, array $data, array $options, Bub
 
         case 'replyto-email':
           $hash = $activity->get('field_activity_hash')->getString();
-          $replacements[$original] = \Drupal::service('activity_send_email.replyto')->getAddress( strlen($hash) > 1 ? $hash : NULL );
+          $replacements[$original] = \Drupal::service('activity_send_email.replyto')
+            ->getAddress(strlen($hash) > 1 ? $hash : NULL);
+          break;
+
+        case 'recipient-data':
+          $user_id = $activity->get('field_activity_recipient_user')->target_id;
+          $user = \Drupal::entityTypeManager()->getStorage("user")->load($user_id);
+          $active_profile = \Drupal::entityTypeManager()->getStorage('profile')->loadByUser($user, "organisation_profile");
+
+          // Designate the field we want to render.
+          $field_name = 'field_org_address';
+          $display = [
+            'label' => 'hidden',
+            'type' => 'field_address',
+          ];
+          // Retrieve a render array for that field with the given view mode.
+          $build = $active_profile->$field_name->view($display);
+          // Render the result.
+          $address = \Drupal::service('renderer')->renderRoot($build);
+          $replacements[$original] = \Drupal\Core\Render\Markup::create($active_profile->get('field_org_title')->value . $address);
+          break;
+
+
+        case 'sender-data':
+          $user_id = $activity->get('user_id')->target_id;
+          $user = \Drupal::entityTypeManager()->getStorage("user")->load($user_id);
+          $active_profile = \Drupal::entityTypeManager()->getStorage('profile')->loadByUser($user, "organisation_profile");
+
+          // Designate the field we want to render.
+          $field_names = array('field_org_address', 'field_org_telephone', 'field_org_email');
+          $display = [
+            'label' => 'hidden',
+            'type' => 'field_address',
+          ];
+
+          $data = ""
```

Comment:
This does not seem right. Line is missing semicolon.

Reply:
typo, was in hurry :) thnx